### PR TITLE
docs: Rewrite ingesting usage guide

### PR DESCRIPTION
**Pull request by sarkissianraffi**
## Summary
- Rewrites the `ingesting-usage.mdx` page into a comprehensive integration guide
- Adds event design patterns with industry-specific use cases (AI/LLM, Cloud/GPU, Fintech, API Platforms, Storage/CDN, Marketplace)
- Documents all delivery methods: REST API (single + batch), client libraries (Python, Node.js, Ruby, Go), Kafka/Redpanda, and S3 bulk import
- Covers idempotency, deduplication, edge cases (late arrivals, corrections, pre-subscription events), and event validation
- Uses Mintlify components: Tabs, CodeGroup, AccordionGroup, CardGroup, Mermaid diagrams, Info/Warning/Tip callouts

## Test plan
- [ ] Verify page renders correctly on Mintlify preview (callouts, tabs, code groups, mermaid diagrams)
- [ ] Check all internal links resolve (`/guide/billable-metrics/create-billable-metrics`, `/guide/plans/overview`, `/guide/events/metering-source`)
- [ ] Confirm the events-structure.png image displays in the "Designing your events" section

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: guide/events/usage-performances.mdx [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+
+### Kafka / Redpanda configuration
+
+Kafka ingestion requires the **ClickHouse event store** (see Architecture section above).
+
+<AccordionGroup>
+  <Accordion title="Lago Cloud">
+    Lago Cloud includes a managed Kafka endpoint. Contact the team for your connection details (broker address, credentials, topic name). No infrastructure to deploy.
+  </Accordion>
+  <Accordion title="Self-hosted setup">
```
Comment (jdenquin) [inline guide/events/usage-performances.mdx]: This is not totally true, I suggest to remove this for now

**Review by jdenquin — APPROVED** [review]
